streams/src/test/java/org/apache/kafka/streams/integration/KafkaStreamsTelemetryIntegrationTest.java

```diff
@@ -0,0 +1,533 @@
+        final Properties clientProps = new Properties();
+        clientProps.put("bootstrap.servers", cluster.bootstrapServers());
+        final Topology topology = complexTopology();
+        try (final KafkaStreams streams = new KafkaStreams(topology, properties);
+             final ClientMetricsCommand.ClientMetricsService clientMetricsService = new ClientMetricsCommand.ClientMetricsService(clientProps)) {
+            IntegrationTestUtils.startApplicationAndWaitUntilRunning(streams);
+
+            final ClientInstanceIds clientInstanceIds = streams.clientInstanceIds(Duration.ofSeconds(60));
+            final Uuid adminInstanceId = clientInstanceIds.adminInstanceId();
+            final Uuid mainConsumerInstanceId = clientInstanceIds.consumerInstanceIds().entrySet().stream()
+                    .filter(entry -> !entry.getKey().contains("restore"))
+                    .map(Map.Entry::getValue)
+                    .findFirst().get();
+            assertNotNull(adminInstanceId);
+            assertNotNull(mainConsumerInstanceId);
+
+            final String[] metricsSubscriptionParameters = new String[]{"--bootstrap-server", cluster.bootstrapServers(),  "--metrics", "org.apache.kafka.stream",
+                                                                        "--alter", "--name", mainConsumerInstanceId.toString(), "--interval", "1000"};
+            final ClientMetricsCommand.ClientMetricsCommandOptions commandOptions = new ClientMetricsCommand.ClientMetricsCommandOptions(metricsSubscriptionParameters);
+            clientMetricsService.alterClientMetrics(commandOptions);
+            final List<String> words = Arrays.asList("foo", "bar", "baz", "all", "streams", "lead", "to", "kafka");
+            clientProps.put(ProducerConfig.KEY_SERIALIZER_CLASS_CONFIG, ByteArraySerializer.class);
+            clientProps.put(ProducerConfig.VALUE_SERIALIZER_CLASS_CONFIG, StringSerializer.class);
+
+            IntegrationTestUtils.produceValuesSynchronously(inputTopicTwoPartitions,
+                    words,
+                    clientProps,
+                     cluster.time);
+
+            clientProps.put(ConsumerConfig.KEY_DESERIALIZER_CLASS_CONFIG, StringDeserializer.class);
+            clientProps.put(ConsumerConfig.VALUE_DESERIALIZER_CLASS_CONFIG, LongDeserializer.class);
+            clientProps.put(ConsumerConfig.GROUP_ID_CONFIG, "test-consumer-group");
+
+            final List<KeyValue<String, Long>> actualKeyValues =  IntegrationTestUtils.waitUntilMinKeyValueRecordsReceived(clientProps, outputTopicTwoPartitions, words.size());
+            assertEquals(words.size(), actualKeyValues.size());
+
+            TestUtils.waitForCondition(() -> SUBSCRIBED_CLIENT_METRIC_DATA_COUNT.get(mainConsumerInstanceId) >= 10_000,
+                    30_000 * 10, //Temporary until a workaround for getting past the initial push interval of 5 minutes then set at 30 seconds
+                    "Never received subscribed metrics");
```

Comment:
As the comment indicates, this is temporary until the metrics push timing can be worked out.